sys/security/mac_do/mac_do.c

```diff
@@ -1883,18 +1883,15 @@ grant_supplementary_group_from_flags(const flags_t flags)
 }
 
 static int
-rule_grant_supplementary_groups(const struct rule *const rule,
-    const struct ucred *const old_cred, const struct ucred *const new_cred)
+rule_grant_setgroups(const struct rule *const rule,
+    const struct ucred *const old_cred, const int ngroups,
+    const gid_t *const groups)
 {
-	const gid_t *const old_groups = old_cred->cr_groups;
-	const gid_t *const new_groups = new_cred->cr_groups;
-	const int old_ngroups = old_cred->cr_ngroups;
-	const int new_ngroups = new_cred->cr_ngroups;
 	const flags_t gid_flags = rule->gid_flags;
 	const bool current_has_supp = (gid_flags & MDF_CURRENT) != 0 &&
 	    (gid_flags & MDF_SUPP_MASK) != 0;
 	id_nb_t rule_idx = 0;
-	int old_idx = 1, new_idx = 1;
+	int old_idx = 1, new_idx = 0;
```

Comment:
Should be:

```suggestion
	int old_idx = 0, new_idx = 0;
```

after recent changes in -CURRENT.  Will fix that in a separate commit (tomorrow probably).